units/task_monitor_execution_timeout.go

```diff
@@ -198,6 +198,13 @@ func cleanUpTimedOutTask(t *task.Task) error {
 		}
 		return errors.Wrap(model.MarkEnd(t, "monitor", time.Now(), detail, false, &model.StatusChanges{}), "error marking task ended")
 	}
+	if t.IsPartOfSingleHostTaskGroup() {
+		if err = t.SetResetWhenFinished(); err != nil {
+			return errors.Wrap(err, "can't mark task group for reset")
+		}
+		return errors.Wrap(model.MarkEnd(t, "monitor", time.Now(), detail, false, &model.StatusChanges{}), "error marking task ended")
+	}
```

Comment:
does it make sense to try and fold this into `model.TryResetTask`? Is there a case where we don't want TryResetTask to do this?

Reply:
Actually yeah good catch, TryResetTask already does this.